pkgs/applications/misc/st/default.nix

```diff
@@ -1,33 +1,32 @@
-{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft, fontconfig
-, conf ? null, patches ? []}:
+{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft
+, fontconfig, conf ? null, patches ? []}:
 
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  name = "st-0.6";
-  
+let patches' = if isNull patches then [] else patches;
```

Comment:
The recursive issue is obvious, but having two ways of saying "no additional patches" seems odd to me. Is there a benefit to passing `null` over simply `[]`?